File: PL031.cpp
#include <iostream>
#include <bits/stdc++.h>
using namespace std;

vector<int> getMoreAndLess(int arr[], int n, int x)
{

    int lessNum = 0;
    int maxNum = 0;

    for (int i = 0; i < n; i++)
    {
        if (arr[i] < x)
        {
            lessNum++;
        }
        else if (arr[i] > x)
        {
            maxNum++;
        }
        else
        {
            lessNum++;
            maxNum++;
        }
    }

    vector<int> v;
    v.push_back(lessNum);
    v.push_back(maxNum);
    return v;
}

int main()
{
    int n = 6;
    int arr[] = {1, 2, 34, 34, 72, 91};
    int x = 34;

    vector<int> v;

    v = getMoreAndLess(arr, n, x);

    cout << "Number of elements smaller than " << x << " : " << v[0] << endl;
    cout << "Number of elements greater than " << x << " : " << v[1] << endl;

    return 0;
}

File: PL031.java
import java.util.ArrayList;
import java.util.List;

public class PL031 {
    public static void main(String[] args) {
        int n = 6;
        int x = 314;
        int[] arr = { 1, 2, 34, 34, 72, 91 };
        List<Integer> list = new ArrayList<>();
        list = getMoreAndLess(arr, n, x);
        System.out.println("The number of smaller and equal elements are : " + list.get(0));
        System.out.println("The number of greater and equal elements are : " + list.get(1));
    }

    public static List<Integer> getMoreAndLess(int[] arr, int n, int x) {

        int lessNum = 0;
        int maxNum = 0;

        for (int i = 0; i < n; i++) {
            if (arr[i] < x) {
                lessNum++;
            } else if (arr[i] > x) {
                maxNum++;
            } else {
                lessNum++;
                maxNum++;
            }
        }

        List<Integer> result = new ArrayList<>();
        result.add(lessNum);
        result.add(maxNum);

        return result;
    }
}
